### ISSUE TITLE
Peptide-Shaker errors

### ISSUE THREAD

**Issue by Lichun-bio** [compomics/peptide-shaker#399]
Hi there,

Could you please help me with two errors (below) about Peptide Shaker?  Thank you very much!

Lichun
__________________________________________________________________________________

**1st:**

Sun Apr 12 01:26:02 CDT 2020: PeptideShaker version 1.16.44.
Memory given to the Java virtual machine: 27962376192.
Total amount of memory in the Java virtual machine: 128974848.
Free memory: 94896128.
Java version: 1.8.0_211.
java.lang.NullPointerException
	at com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex.getProteinMappingWithoutVariants(FMIndex.java:4155)
	at com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex.getProteinMapping(FMIndex.java:3827)
	at com.compomics.util.experiment.identification.protein_inference.fm_index.FMIndex.getProteinMapping(FMIndex.java:3801)
	at eu.isas.peptideshaker.protein_inference.TagMapper.mapTagsForSpectrumMatch(TagMapper.java:192)
	at eu.isas.peptideshaker.protein_inference.TagMapper.access$000(TagMapper.java:54)
	at eu.isas.peptideshaker.protein_inference.TagMapper$SpectrumMatchTagMapperRunnable.run(TagMapper.java:380)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
_____________________________________________________________________________________________________
2nd:

Sat Apr 11 11:04:06 CDT 2020: PeptideShaker version 1.16.44.
Memory given to the Java virtual machine: 27962376192.
Total amount of memory in the Java virtual machine: 128974848.
Free memory: 95046240.
Java version: 1.8.0_211.
Exception in thread "TimerQueue" java.lang.OutOfMemory**Error: GC overhead limit exceeded**
java.lang.NullPointerException
	at eu.isas.peptideshaker.validation.MatchesValidator$ProteinValidatorRunnable.run(MatchesValidator.java:1933)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Exception in thread "AWT-EventQueue-0" java.lang.OutOfMemoryError: GC overhead limit exceeded
java.sql.SQLException: Java exception: 'GC overhead limit exceeded: java.lang.OutOfMemoryError'.
	at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.Util.javaException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedConnection.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.ConnectionChild.handleException(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeQuery(Unknown Source)
	at org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getTables(Unknown Source)
	at com.compomics.util.db.ObjectsDB.getTables(ObjectsDB.java:285)
	at com.compomics.util.db.ObjectsDB.hasTable(ObjectsDB.java:259)
	at com.compomics.util.experiment.identification.IdentificationDB.loadAssumptions(IdentificationDB.java:715)
	at com.compomics.util.experiment.identification.Identification.loadAssumptions(Identification.java:174)
	at com.compomics.util.experiment.identification.matches_iterators.PsmIterator.checkBuffer(PsmIterator.java:288)
	at com.compomics.util.experiment.identification.matches_iterators.PsmIterator.next(PsmIterator.java:231)
	at eu.isas.peptideshaker.ptm.PtmScorer.scorePTMs(PtmScorer.java:808)
	at eu.isas.peptideshaker.ptm.PtmScorer.scorePeptidePtms(PtmScorer.java:1469)
	at eu.isas.peptideshaker.PeptideShaker.processIdentifications(PeptideShaker.java:490)
	at eu.isas.peptideshaker.fileimport.FileImporter$IdProcessorFromFile.importFiles(FileImporter.java:547)
	at eu.isas.peptideshaker.fileimport.FileImporter.importFiles(FileImporter.java:151)
	at eu.isas.peptideshaker.PeptideShaker.importFiles(PeptideShaker.java:229)
	at eu.isas.peptideshaker.cmd.PeptideShakerCLI.createProject(PeptideShakerCLI.java:748)
	at eu.isas.peptideshaker.cmd.PeptideShakerCLI.call(PeptideShakerCLI.java:181)
	at eu.isas.peptideshaker.cmd.PeptideShakerCLI.main(PeptideShakerCLI.java:957)
Caused by: java.sql.SQLException: Java exception: 'GC overhead limit exceeded: java.lang.OutOfMemoryError'.
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)
	at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(Unknown Source)
	... 26 more
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
_________________________________________________________________________________________________________

**Comment by hbarsnes** [comment]
Hi Lichun,

It looks like you are trying to load do novo sequencing data is that correct? Can you tell us a bit more about the data you are trying to load?

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,

Thanks for your quick response.

I just started to use SearchGUI, and trying to see which engine (or
combined engines) could give us better results. I used one or more
proteomics search engines, both de novo engines of Novor and DirecTag, and
Peptide Shaker in SerachGUI to process a raw data file.

[image: image.png]

Below is the *SearchGUI report* part of two errors about Peptides shaker:
______________________________________________________________________
*1st:*
Sun Apr 12 01:26:03 CDT 2020        Unzipping searchgui_out.zip.
Sun Apr 12 01:26:04 CDT 2020        Import process for
Exp1_T0_C5-7_initial-Andromeda (Sample: Exp1_T0_C5-7_initial-Andromeda,
Replicate: 0)


Sun Apr 12 01:26:04 CDT 2020        Importing sequences from
dbVislova-Ottesen.deduped.trim_concatenated_target_decoy.fasta.
Sun Apr 12 01:42:11 CDT 2020        FASTA file import completed.
Sun Apr 12 01:42:11 CDT 2020        Importing gene mappings.
Sun Apr 12 01:45:33 CDT 2020        Establishing local database connection.
Sun Apr 12 01:45:34 CDT 2020        Reading identification files.
Sun Apr 12 01:45:34 CDT 2020        Parsing Exp1_T0_C5-7_initial.tags.
Sun Apr 12 01:45:35 CDT 2020        Loading spectra for
Exp1_T0_C5-7_initial.tags.
Sun Apr 12 01:45:35 CDT 2020        Importing Exp1_T0_C5-7_initial.mgf
Sun Apr 12 01:45:35 CDT 2020        Exp1_T0_C5-7_initial.mgf imported.
Sun Apr 12 01:45:36 CDT 2020        Mapping de novo tags to peptides.
Sun Apr 12 01:45:38 CDT 2020        An error occurred: null
Sun Apr 12 01:45:38 CDT 2020        Please contact the developers (
https://github.com/compomics/peptide-shaker/issues).
__________________________________________________________________________________________________
*2nd:*

Sat Apr 11 10:56:17 CDT 2020        Novor finished for
EColi-OldTrypsin-Original-1.mgf (6 minutes 25.0 seconds).

Sat Apr 11 10:56:18 CDT 2020        Processing
EColi-OldTrypsin-Original-1.mgf with DirecTag.
..................................

Finished sequence tagging spectra; 193.829 seconds elapsed.
Mean original (filtered) peak count: 870.724 (99.9685)
Min/Max original (filtered) peak count: 26/1667 (26/100)
Filtered out 88.5189% of peaks before tagging.
Overall stats: 85108 spectra; 34645279 tag graph edges; 594937104 tags
generated; 824924 tags retained
Writing tags to "EColi-OldTrypsin-Original-1.tags".
Finished file "E:\Projects\April-Search
GUI\EColi-OldTrypsin-Original-1.mgf"; 314.345 seconds elapsed.
Finished tagging 1 files; 314.345 seconds elapsed.


Sat Apr 11 11:01:34 CDT 2020        DirecTag finished for
EColi-OldTrypsin-Original-1.mgf (5 minutes 16.0 seconds).

Sat Apr 11 11:01:34 CDT 2020        Zipping output files.

Sat Apr 11 11:04:05 CDT 2020        Processing identification files
with PeptideShaker.
Sat Apr 11 12:27:53 CDT 2020        PeptideShaker processing failed.
See the PeptideShaker log for details.
Sat Apr 11 12:27:53 CDT 2020        Search Canceled!
Sat Apr 11 16:05:20 CDT 2020        The search or processing did not
finish properly!
-----------------------------------------------------------------Please
let me know if you need more information.

Thank you very much

Lichun

**Comment by Lichun-bio** [comment]
Hi Harald,

I ran another search yesterday. The program seems to be frozen. Below are two screenshots (taken after >20hr searching) of SerachGUI and PeptideShaker status windows :

![image](https://user-images.githubusercontent.com/63607025/79228667-7762a800-7e27-11ea-9c4f-3468ea193c8f.png)

![image](https://user-images.githubusercontent.com/63607025/79228746-96f9d080-7e27-11ea-94bf-5d23a48bf55c.png)

Thanks!
Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

How big is your sequence database? Mapping sequence tags to very large database can take a long time. Providing more memory is usually the trick, but it seems like you have already given PeptideShaker 28 GB?

You could try closing SearchGUI and loading the output separately in PeptideShaker?

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,

Yes, 2 databases are big. Here is the size of one of them:
dbVislova-Ottesen.deduped.trim_concatenated_target_decoy.fasta, 1 541 526
KB (dbVislova-Ottesen.deduped.trim.fasta: 699 893 KB). How long does it
usually take to mapping sequence tags to such a large size database?

PeptideShaker also has problem (if it's not taking more than 8 hours to
search) with one small databases (Ecoli)  I used. The size is not very big,
Ecoli-04092020_GCF_000005845-2_ASM584v2_protein_concatenated_target_decoy.fasta,
3431 KB (Ecoli-04092020_GCF_000005845-2_ASM584v2_protein.fasta, 1703 KB)

Thanks

Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

> How long does it usually take to mapping sequence tags to such a large size database?

To be honest, I do not really know. We generally work with the human sequences from UniProt, which even with the reverse decoys added is just around 26 MB. And usually with normal searches, i.e. not tags.

I'm not sure how big databases we have tested. I'd have to check with the developers in charge of this part of the code. @dominik-kopczynski @mvaudel Do you guys have any comments or suggestions?

> PeptideShaker also has problem (if it's not taking more than 8 hours to search) with one small databases (Ecoli) I used.

No, in this case it should not take eight hours. Where does it stop? And are there any errors in the log file?

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,

Here is the SearchGUI log of the failed Ecoli sample search. I ran all 8
proteomics search engines, 2 de novo engines and Peptide Shaker. There was
no problem until it started PeptideShaker.
_________________________________________________________________

Fri Apr 10 02:03:30 CDT 2020        DirecTag finished for
EColi-OldTrypsin-Original-1.mgf (5 minutes 9.0 seconds).
Fri Apr 10 02:03:30 CDT 2020        Zipping output files.
Fri Apr 10 02:05:58 CDT 2020        Processing identification files
with PeptideShaker.

Fri Apr 10 08:14:33 CDT 2020        PeptideShaker finished for
Ecoli-oldTrypsin-8Engine (sample: Ecoli-oldTrypsin-8Engine, replicate:
0) (6 hours 8 minutes 35.0 seconds).
Fri Apr 10 08:14:36 CDT 2020        Search Completed (7 hours 40
minutes 38.0 seconds).
__________________________________________________________________

I left it running overnight, and thought the program was frozen and
canceled it next morning. Peptide Shaker's log does not show there was
any problem:
_______________________________________________________

Fri Apr 10 02:05:59 CDT 2020: PeptideShaker version 1.16.44.

Memory given to the Java virtual machine: 27962376192.

Total amount of memory in the Java virtual machine: 128974848.

Free memory: 94896128.

Java version: 1.8.0_211.


Fri Apr 10 08:42:57 CDT 2020: PeptideShaker version 1.16.44.

Memory given to the Java virtual machine: 27962376192.

Total amount of memory in the Java virtual machine: 128974848.

Free memory: 124461312.

Java version: 1.8.0_211.

_________________________________________________________

Regarding the database size, the other field sample database we use
with the reverse decoys added is even larger at 5 413 600 KB. It seems
there is more trouble to search with it. I am still working on it and
will let you know if we have
comments or suggestions.

Thank you very much for your time and help.

Regards

Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

Would it be possible for you to share the search results for the small database so that we can try to figure out what is going on?

For the large database, I'm still waiting on input from the developers of that part of the code.

Best regards,
Harald

**Comment by dominik-kopczynski** [comment]
Hi Lichun,

the mapping function was designed in a way that it can manage databases as long as they fit into the memory. But the error of your first issue looks more like a bug in the mapper than a memory problem. To make a better assessment of what's going wrong, it would be really great, if we could get some parts of your data to test it on our machines. Is it possible to provide us some data? 

Cheers,
Dominik

**Comment by Lichun-bio** [comment]
Hi Harald,

Here is the results from the small database search.

I tried to post the files @
https://github.com/compomics/peptide-shaker/issues/399. But Zip file is too
big and some unzipped file types are not accepted. Hope you can open the
file in the google drive. Please let me know if you need more information.

Regards

Lichun

---------- Forwarded message ---------
From: Lichun Zhang <zhanglc@gmail.com>
Date: Mon, Apr 20, 2020 at 9:30 AM
Subject: SearchGUI files
To: Lichun Zhang <zhanglc@gmail.com>


 searchgui_out.zip
<https://drive.google.com/file/d/19KhKwwpyWNK-S8LHL-ld3GIXnCFKKkkQ/view?usp=drive_web>

**Comment by Lichun-bio** [comment]
HOT311_Exp1_T0_C5-7_initial.zip
<https://drive.google.com/file/d/166ZVKaET54LrN8c4Uq1YMNJ2IBIPnTz3/view?usp=drive_web>
 dbVislova-Ottesen.deduped.trim.zip
<https://drive.google.com/file/d/1EcPAUQV2I5O4fG4RryK8-j9jdPpLc47F/view?usp=drive_web>
Hi Dominik,

Here is the raw data file and the database without the reverse decoys
added. I used the reverse decoys added database for the searching. Please
let me know if you  have trouble to open the file in the google drive.

Thanks

Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

Thanks for sharing the results from the small database search. However, I will also need the mgf and the FASTA files in order to be able to load the data in PeptideShaker.

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,  Here they are. One is attached, one is in Google Drive. ~Lichun
 EColi-OldTrypsin-Original-1.zip
<https://drive.google.com/file/d/165edWpQajEwDFPdMKpuZI-kek-iuU2Nz/view?usp=drive_web>

**Comment by hbarsnes** [comment]
> One is attached

I'm afraid that does not work when replying to issues via email. If that is what you tried?

**Comment by Lichun-bio** [comment]
Yes, I replied with email. Here is the file I tried to attach.
[Ecoli-04092020_GCF_000005845-2_ASM584v2_protein_concatenated_target_decoy.zip](https://github.com/compomics/peptide-shaker/files/4505952/Ecoli-04092020_GCF_000005845-2_ASM584v2_protein_concatenated_target_decoy.zip)

**Comment by hbarsnes** [comment]
Thanks! I got all the files I need now. The first thing I noticed is that while your database is indeed not very big, your mgf file is. Have you tried loading the results of a single search engine on it's own? Generally, the results will be more or less the same, and the processing should go a lot quicker.

Also, with mgf files this big, I'd recommend staying away from OMSSA. It was not really created for big mgf files, and as you can see the size of the output file is very large as well. So you'll save both time and space by excluding it.

**Comment by Lichun-bio** [comment]
Hi Harald,

Thanks for the recommendation! Yes, I did search with each single engine
after the eight engine search had failed. Those single engine search did
went through.

Regards
Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

> Thanks for the recommendation! Yes, I did search with each single engine after the eight engine search had failed. Those single engine search did went through.

I'm able to open the X! Tandem results without any issues as well. I think that to load all of them I'd need something more powerful than my laptop in order for it to not take a very long time.

I'd therefore recommend staying with one or two search engines when having mgf files that are this big, unless you really need to have all of them. Is that ok?

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,

Do you have a recommendation which engine or combination to use for
searching? Or how to pick which engine/engines to search? Our mgf data
files usually are larger than 1GB.

Thanks
Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

I generally recommend X! Tandem. In our experience it usually performs the best and covers almost all of the hits from the other search engines.

But the best approach is to try a couple of them on your own data and see which seem to perform the best. And then use those (or just the single one) for your future searches.

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harald,

Great! Thanks for the advice.

Regards
Lichun

**Comment by Lichun-bio** [comment]
Hi Harald and Dominik,

Just wondering if you got a chance to test the big raw data file and
database I sent (via the google drive) on your machine?  I tried with the
upgraded SearchGUI and got the same error.

Thank you

Lichun

**Comment by dominik-kopczynski** [comment]
Hi Lichun,

I am sorry but due to a mistake of mine, I don't have your files any more on my computer. Can you please share them once again with me? 

Thank you

**Comment by Lichun-bio** [comment]
dbVislova-Ottesen.deduped.trim.zip
<https://drive.google.com/file/d/1dboC-NHjDCozuWUq6KZkwKiTfccgBXVD/view?usp=drive_web>
 HOT311_Exp1_T0_C5-7_initial.zip
<https://drive.google.com/file/d/1LjArn_YEgHjlU4jtWxHYiw0rTYcYAs9H/view?usp=drive_web>
Hi Dominik,

Shared files in the Google drive are the raw data file and the
database without the reverse decoys added. I used the reverse decoys added
database for the searching.

Please let me know if you have trouble to get files.

Thanks

Lichun

**Comment by dominik-kopczynski** [comment]
Hi Lichun,

Thank you for the response. I requested permissions for the files.

Cheers,
Dominik

**Comment by Lichun-bio** [comment]
Hi Dominik,

I have added your email to the shared list. Please let me know if you have
trouble to access files.

Thank you

Lichun

**Comment by dominik-kopczynski** [comment]
Thank you, I am downloading the two files. Will have a look afterwards.

**Comment by dominik-kopczynski** [comment]
Btw: how much memory does your computer have? And is it Win/Linux/Mac?

**Comment by Lichun-bio** [comment]
Hi Dominik,

Here is the information about the computer running SearchGUI and
PeptideShaker.

[image: image.png]

**Comment by Lichun-bio** [comment]
![image](https://user-images.githubusercontent.com/63607025/83045232-f04e4580-a00a-11ea-9514-6fe949c15d8b.png)

**Comment by Lichun-bio** [comment]
Hi Dominik,

Could you please update me how the test of the big raw data file and database I sent goes if you got a chance to do it on your machine?  

Thank you

Lichun

**Comment by hbarsnes** [comment]
This issue should hopefully be fixed in the new release that is available now. If not, please let us know by opening a new issue.

**Comment by Lichun-bio** [comment]
Hi Harald,

Thanks for letting me know.

I tried the newest version SearchGUI 4.0.1, but could not find the chemical
modification of CarbamOxC (delta mass=73.016). Could you please help to
check if I missed something?

Thanks again
Lichun

**Comment by hbarsnes** [comment]
Hi Lichun,

It is indeed not included in the list of default modifications, but you can always add it yourself as a user modification. In SearchGUI click Edit > Modifications at the top of the main dialog, then click the Add button below the User Modifications table.

Best regards,
Harald

**Comment by Lichun-bio** [comment]
Hi Harals,

Thanks for the quick response and instructions. I'll add it as a user
modification.

Cheers
Lichun
